openid-4-verifiable-presentations-1_0.md

```diff
@@ -616,11 +622,322 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+the request to the Wallet does not have access to the claim value. Therefore, Verifiers
+must treat restrictions expressed using `values` as a best-effort way to improve
+user privacy, but MUST NOT rely on it for security checks.
+
+The purpose of the `claim_sets` syntax is to provide a way for a 
+verifier to describe alternative ways a given credential can satisfy the request. 
+The array ordering expresses the Verifier's preference for how the request must 
+be fulfilled. The first element in the array is the most preferred and the last 
+element in the array is the least preferred. Verifiers SHOULD use the principle of 
+least information disclosure to influence how they order these options. For example, a 
+proof of age request should prioritize requesting an attribute like `age_over_18` over
+an attribute like `birth_date`.
+
+If the Wallet cannot deliver all claims requested by the Verifier
+according to these rules, it MUST NOT return the respective Credential.
+
+#### Selecting Credentials
+
+The following rules apply for selecting Credentials via `credentials` and `credential_sets`:
+
+- If `credential_sets` is not provided, the Verifier requests all
+  Credentials in `credentials` to be returned.
+- Otherwise, the Verifier requests all of the Credential Set Queries in the `credential_sets` array
+  with `required` evaluated as true to be returned at a minimum and optionally any of the credential set queries 
+  with `required` evaluated as false.
+
+For each credential set query inside the `credential_sets` array, in order to
+satisfy the query, the Wallet MUST return a credential or credentials that
+match to one of the `options` inside the object. It is typically
+expected that the Wallet presents the End-User with a choice of which
+credential to present if multiple credentials match the query.
+
+Credentials not matching the respective constraints expressed within
+`credentials` MUST NOT be returned, i.e., they are treated as if
+they would not exist in the Wallet.
+
+If the Wallet cannot deliver all non-optional Credentials requested by the
+Verifier according to these rules, it MUST NOT return any credential(s).
+
+#### User Interface Considerations {#vp_query_ui}
+
+While this specification provides the mechanisms for requesting different sets
+of claims and credentials, it does not make assumptions about the user interface
+of the Wallet, for example, if users can select which combination of credentials
+to present.
+
+#### Security Considerations {#vp_query_security}
+
+While the Verifier can specify various constraints both on the claims level and
+the credential level as shown above, it MUST NOT rely on the Wallet to enforce
+these constraints. The Wallet is not controlled by the Verifier and the Verifier
+MUST perform its own security checks on the returned credentials and
+presentations.
+
+## Format-specific Properties {#format_specific_properties}
+
+The following format-specific properties are defined:
```

Comment:
```suggestion
OpenID for Verifiable Presentations is Credential Format agnostic, i.e., it is designed to allow applications to request and receive Verifiable Presentations and Verifiable Credentials in any Credential Format. This section defines Credential Format Profiles for some of the known Credential Formats. Other specifications or deployments can define their own Credential Format Profiles:
```

Reply:
one option would be to move these credential format specific parts to annex B, but that should be another issue/PR.